### PR TITLE
Alt-clicking defibrillator removes cell

### PR DESCRIPTION
**Pull request by norill**
## About The Pull Request
- added alt-clicking defibrillator to remove cell
- fixed various charge-related bugs of defib and defib wall mounts

## Why It's Good For The Game
doctors dont carry screwdrivers and robbing dead engis is annoying

## Changelog
:cl:
qol: You can now remove power cell from defibrillator by alt-clicking it.
fix: Fixed defibrillator paddles and wall mounts not detecting charge changes.
/:cl:

### CODE REVIEW

**Comment by JohnFulpWillard** [comment]
Why not right click?

File: code/game/objects/items/defib.dm [hunk 1]
```diff
@@ -139,6 +147,17 @@
...
 
+/obj/item/defibrillator/AltClick(mob/living/carbon/user)
+	if(!istype(user) || !user.canUseTopic(src, BE_CLOSE))
+		return
+
+	if(cell)
+		cell.update_appearance()
+		user.put_in_hands(cell)
+		cell = null
+		to_chat(user, "<span class='notice'>You remove the cell from [src].</span>")
```
Comment (Mothblocks) [inline code/game/objects/items/defib.dm:158]:
```suggestion
		balloon_alert(user, "cell removed")
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Right-click.

**Comment by norill** [comment]
> Why not right click?

consistency. alt-clicks are used for removing things from things in many places. including removing defibs from wall mounts

**Comment by Mothblocks** [comment]
Fair enough, I'll let it be on alt-click for now then.

**Review by Mothblocks — APPROVED** [review]

**Comment by optimumtact** [comment]
No, we should change the consistency, not stick with a dumb undiscoverable default

**Comment by Mothblocks** [comment]
I'd rather change all alt-click-to-remove in one swoop rather than have some be right-click and some be alt-click, when the whole goal of the RMB changes is for a player to be able to consistently figure what does what.

**Comment by maxymax13** [comment]
> I'd rather change all alt-click-to-remove in one swoop rather than have some be right-click and some be alt-click, when the whole goal of the RMB changes is for a player to be able to consistently figure what does what.

...by making rmb the new "mystery button that does basically anything"?

**Comment by Mothblocks** [comment]
To the secondary action button, yes.

**Comment by Fikou** [comment]
doctors literally get a screwdriver in the same locker they get defibs in, remove those screwdrivers if you are gonna remove that functionality

**Comment by Mothblocks** [comment]
*sweats*

**Comment by optimumtact** [comment]
honestly this PR is stupid, it's not hard to get screwdriver.
I changed my mind I don't like it anymore

**Comment by Mothblocks** [comment]
Actually, I kinda blanked on that as well, I also don't really care about this anymore :sweat_smile:

**Comment by Hamurlik** [comment]
just make it both alt-click and right-click those 2 spots are free and 2 camps will be happy

**Comment by norill** [comment]
> No, we should change the consistency, not stick with a dumb undiscoverable default

> I'd rather change all alt-click-to-remove in one swoop rather than have some be right-click and some be alt-click, when the whole goal of the RMB changes is for a player to be able to consistently figure what does what.

im fine with changing whatever. but first you should come up with a design document of sorts for UX interactions to guide coders on how to assign buttons to interactions, right now it's too messy and inconsistent. as for the "undiscoverability" - from the lack of better options i put interaction information in examine text. in this case it says that alt-click removes the cell upon examination.

> honestly this PR is stupid, it's not hard to get screwdriver.
> I changed my mind I don't like it anymore

> doctors literally get a screwdriver in the same locker they get defibs in, remove those screwdrivers if you are gonna remove that functionality

you only get screwdrivers in med storage on meta and tram, other stations dont have them. and last time i played on meta all the shit from storage got bluespaced before i joined. you can print defibs and basic cells in med techfab, but you cant print a damn screwdriver. i ended up printing a new defib every time i needed it.

besides, my main point is that doctors shouldnt need a screwdriver. roles are there for a reason


> just make it both alt-click and right-click those 2 spots are free and 2 camps will be happy

im fine with that. and btw, i didnt remove the screwdriver thing for "backward compatibility" - people who are used to it would have a grace period to learn the screwdriver-less method

also if you gonna close it - this PR has a few fixes besides that screwdriver thing, should i move them to a new PR or remove the screwdriver thing from this one?

**Comment by Mothblocks** [comment]
> also if you gonna close it - this PR has a few fixes besides that screwdriver thing, should i move them to a new PR or remove the screwdriver thing from this one?

Move to a new PR.

**Pull request closed without merging**